scripts/falco-driver-loader

```diff
@@ -420,18 +480,100 @@ fi
 DRIVER_VERSION="@PROBE_VERSION@"
 DRIVER_NAME="@PROBE_NAME@"
 
+DRIVER="module"
+if [ -v FALCO_BPF_PROBE ]; then
+	DRIVER="bpf"
+fi
+
+ENABLE_COMPILE=
+ENABLE_DOWNLOAD=
+SKIP_LOAD=
+
+print_usage() {
+	echo ""
+	echo "Usage:"
+	echo "  falco-driver-loader [driver] [options]"
+	echo ""
+	echo "Available drivers:"
+	echo "  module        kernel module (default)"
+	echo "  bpf           eBPF probe"
+	echo ""
+	echo "Options:"
+	echo "  --help        show brief help"
+	echo "  --compile     try to compile the driver locally"
+	echo "  --download    try to download a prebuilt driver"
+	echo "  --skip-load   do not load the driver"
+	echo ""
+}
+
+has_args=
+has_opts=
+
+while test $# -gt 0; do
+  case "$1" in
+	module|bpf)
+	  if [ -n "$has_args" ]; then
+	  	>&2 echo "Only one driver can be passed"
+	  	print_usage
+		  exit 1
+	  else
+      DRIVER="$1"
+	    has_args="true"
+      shift
+    fi
+      ;;
+    -h|--help)
+      print_usage
+      exit 0
+      ;;
+    --compile)
+      	ENABLE_COMPILE="yes"
+	    has_opts="true"
+      shift
+      ;;
+    --download)
+	    ENABLE_DOWNLOAD="yes"
+	    has_opts="true"
+      shift
+      ;;
+	--skip-load)
+	    SKIP_LOAD="yes"
+      shift
+      ;;
+    --*)
+	  >&2 echo "Unknown option: $1"
+	  print_usage
+	  exit 1
+      ;;
+	*)
+	  >&2 echo "Unknown driver: $1"
+	  print_usage
+	  exit 1
+      ;;
+  esac
+done
+
+if [ -z "$has_opts" ]; then
+    ENABLE_COMPILE="yes"
+	ENABLE_DOWNLOAD="yes"
+fi
+
 if [ "$(id -u)" != 0 ]; then
-	echo "Installer must be run as root (or with sudo)."
+	>&2 echo "This program must be run as root (or with sudo)"
 	exit 1
 fi
 
 if ! hash curl > /dev/null 2>&1; then
```

Comment:
Can you please add a check for `dkms` when user wants to compile?

Reply:
It is already present [here](https://github.com/falcosecurity/falco/pull/1200/files#diff-b1a242527066116a4e7485802c03bbb3R154) (see the `else` branch).
The check will not terminate the execution as like as the previous script version.

Reply:
Furthermore, we need to continue the execution so the download strategy can happen.
When `dkms` does not exist, users will see `* Skipping dkms install (dkms not found)`